src/Common/FailPoint.cpp

```diff
@@ -246,11 +197,11 @@ void FailPointInjection::wait(const String & fail_point_name)
     std::unique_lock lock(mu);
     auto iter = fail_point_wait_channels.find(fail_point_name);
     if (iter == fail_point_wait_channels.end())
-        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Can not find channel for fail point {}", fail_point_name);
-
-    lock.unlock();
-    auto ptr = iter->second;
-    ptr->wait();
```

Comment:
(This could wait forever if the SYSTEM DISABLE FAILPOINT happened between the `find` and the `wait`.)